pkix/cert_auth.go

```diff
@@ -59,6 +59,15 @@ func CreateCertificateAuthority(key *Key, organizationalUnit string, expiry time
 		authTemplate.PermittedDNSDomains = permitDomains
 	}
 
+	if !excludePathlen {
+		if pathlen > 0 {
+			authTemplate.MaxPathLen = pathlen
+			authTemplate.MaxPathLenZero = false
```

Comment:
`MaxPathLenZero` is only checked when `MaxPathLen == 0`. With that in mind, can we simplify this logic? Something like:

```
authTemplate.MaxPathLen = pathlen
if excludePathLen {
  authTemplate.MaxPathLen = -1
}
```

Then `MaxPathLenZero` can remain true and we use the explicit `MaxPathLen = -1` to unset it.

Reply:
Sorry for late to response. I should have went with that in the first place xD. Thank you for pointing out.

Reply:
Already made the change.

Reply:
To be honest, the change I've just made doesn't seem right. Let me see what I can do or if you have any suggestion for  making it better. Please let me know.

Reply:
Added a comment on the current iteration.